File: Ruby/tests/TRNCHolidayValidator_test.rb
require 'minitest/autorun'
require_relative '../TRNCHolidayValidator'

class TRNCHolidayValidatorTest < Minitest::Test
  def setup
    @validator = TRNCHolidayValidator.new
  end

  def test_trnc_fixed_holidays
    assert @validator.holiday?(Date.new(2024, 8, 1))
  end

  def test_trnc_islamic_holidays
    assert @validator.holiday?(Date.new(2024, 9, 15))
  end
end

File: Rust/tests/trnc_holiday_validator_test.rs
use crate::trnc_holiday_validator::TRNCHolidayValidator;
use chrono::NaiveDate;

#[test]
fn test_trnc_fixed_holidays() {
    let validator = TRNCHolidayValidator::new(false);
    assert!(validator.is_holiday(NaiveDate::from_ymd(2024, 8, 1)));
}

#[test]
fn test_trnc_islamic_holidays() {
    let validator = TRNCHolidayValidator::new(false);
    assert!(validator.is_holiday(NaiveDate::from_ymd(2024, 9, 15)));
}
